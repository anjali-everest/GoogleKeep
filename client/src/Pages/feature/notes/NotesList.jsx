import React, { useState, useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import { fetchNotes, updateIsEditingNote, updateEditingNoteId, deleteOneNote, updateDeletingNoteId } from "./notesSlice"
import optionsIcon from "./../../../styles/optionsIcon.jpeg"


export const NotesList = () => {
  const dispatch = useDispatch()
  const notes = useSelector(state => state.notes)
  const noteStatus = useSelector(state => state.notes.status)
  const error = useSelector(state => state.notes.error)
  const [option, setOption] = useState(false)

  const onOptionsClicked = () => setOption(!option)
  const onDeleteNoteClicked = (e) => {
    dispatch(updateDeletingNoteId(e.target.id))
    dispatch(deleteOneNote(e.target.id))
  }

  useEffect(() => {
    if (noteStatus === 'idle') {
      dispatch(fetchNotes())
    }
  }, [noteStatus, dispatch])

  const handleNoteClick = (e) => {
    dispatch(updateIsEditingNote())
    dispatch(updateEditingNoteId(e.target.id))
  }

  let content
  if (noteStatus === 'loading') {
    content = <div className="loader">Loading...</div>
  } else if (noteStatus === 'succeeded') {
    content = notes.notes && notes.notes.length !== 0 && notes.notes.map(note => (
      <div id={note.id} key={note.id}>
        <article className="note-excerpt" key={note.id} id={note.id} onClick={handleNoteClick}>
          <h3 id={note.id}>{note.title}</h3>
          <p id={note.id}>{note.content.substring(0, 100)}</p>
        </article>
        <img src={optionsIcon} className="optionsIcon" alt="noteOptions" onClick={onOptionsClicked} />
        {option && <div>
          <button id={note.id} onClick={onDeleteNoteClicked}>Delete</button>
        </div>}
      </div>
    ))
  } else if (noteStatus === 'failed') {
    content = <div>{error}</div>
  }

  return (
    <section>
      <h2>Notes</h2>
      {content}
    </section>
  )
}