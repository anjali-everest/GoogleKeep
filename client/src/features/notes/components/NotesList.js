import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { selectAllNotes, selectApiStatus, selectApiError } from "./../notesSlice";
import { fetchNotes } from "../thunksHandler";
import { Note } from "./Note";

export const NotesList = () => {
  const dispatch = useDispatch();
  const notes = useSelector(selectAllNotes);
  const noteStatus = useSelector(selectApiStatus);
  const error = useSelector(selectApiError);

  useEffect(() => {
    if (noteStatus === "idle") {
      dispatch(fetchNotes());
    }
  }, [noteStatus, dispatch]);

  return (
    <section className="notes-list">
      {noteStatus === "loading" && <div className="loader">Loading...</div>}
      {noteStatus === "succeeded" &&
        notes &&
        notes.length !== 0 &&
        notes.map((note) => <Note key={note.id} note={note} />)}
      {noteStatus === "failed" && <div>{error}</div>}
    </section>
  );
};
