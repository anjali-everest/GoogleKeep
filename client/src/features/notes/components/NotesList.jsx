import React, { useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { selectAllNotes, selectApiStatus, selectApiError } from "../notesSlice";
import { fetchNotes } from "../thunkHandler";
import { Note } from "./Note";

export const NotesList = () => {
  const dispatch = useDispatch();
  const notes = useSelector(selectAllNotes);
  const noteStatus = useSelector(selectApiStatus);
  const error = useSelector(selectApiError);

  useEffect(() => {
    if (noteStatus === "idle") {
      dispatch(fetchNotes());
    }
  }, [noteStatus, dispatch]);

  return (
    <section className="notes-list">
      {noteStatus === "loading" && <div className="loader">Loading...</div>}
      {noteStatus === "succeeded" &&
        notes.notes &&
        notes.notes.length !== 0 &&
        notes.notes.map((note) => <Note note={note} />)}
      {noteStatus === "failed" && <div>{error}</div>}
    </section>
  );
};
