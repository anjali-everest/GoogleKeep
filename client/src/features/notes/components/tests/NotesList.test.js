import React from "react";
import { shallow } from "enzyme";
import { NotesList } from "../NotesList";
import configureMockStore from "redux-mock-store";
import * as ReactReduxHooks from "react-redux";

const mockStore = configureMockStore();

describe("NotesList tests", () => {
  const store = mockStore({
    NotesLists: [],
    status: "idle",
    error: null,
    editingNotesListId: null,
    isEditingNotesList: false,
    deletingNotesListId: null,
  });

  beforeEach(() => {
    jest
      .spyOn(ReactReduxHooks, "useSelector")
      .mockImplementation(() => store.getState());
    jest
      .spyOn(ReactReduxHooks, "useDispatch")
      .mockImplementation(() => store.dispatch);
  });

  it("Should render without crashing", () => {
    shallow(<NotesList />);
  });

  it("should match the NotesList snapshot", function () {
    const notesList = shallow(<NotesList />);
    expect(notesList).toMatchSnapshot();
  });
});
