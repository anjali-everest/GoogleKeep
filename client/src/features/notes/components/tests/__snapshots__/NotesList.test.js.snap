// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesList tests should match the NotesList snapshot 1`] = `ShallowWrapper {}`;
